dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilderUtility.h

```diff
@@ -1,13 +1,16 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-#pragma once
 #include "pch.h"
 #include "winrt/Microsoft.Windows.AppNotifications.Builder.h"
+#include <algorithm>
 
 constexpr size_t c_maxAppNotificationPayload{ 5120 };
 constexpr uint8_t c_maxTextElements{ 3 };
 constexpr uint8_t c_maxButtonElements{ 5 };
+constexpr PCWSTR c_encodedPercent{ L"%25" };
```

Comment:
You could add the encodings in a map and directly translate them by inferencing the map. That way, you don't have to define new constants every single time for each encoding.

I would also suggest that you add all the % encodings here to the map to be thorough:
https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding

Reply:
I think a map would work best here, but only ; % = need to be encoded. These are the only special characters that have a special behavior inside the xml. The link above is referring to URL encoding which in our case, only use 3.
; separates each key/value pair
= separates the key and value
% since we use % encoding, we encode % to retain the raw value

Will change to map construct.